Nef_3/include/CGAL/Nef_3/SNC_point_locator.h

```diff
@@ -849,7 +849,24 @@ class SNC_point_locator_by_spatial_subdivision :
           _CGAL_NEF_TRACEN("found on facet...");
           return make_object(f);
         }
-        if( is.does_intersect_internally(s,f,ip)) {
+
+        // We next check if v is a vertex on the face to avoid a geometric test
+        bool v_vertex_of_f = false;
+        Halffacet_cycle_iterator fci;
+        for(fci=f->facet_cycles_begin(); fci!=f->facet_cycles_end(); ++fci) {
+          if(fci.is_shalfedge()) {
+            SHalfedge_around_facet_circulator sfc(fci), send(sfc);
+            CGAL_For_all(sfc,send) {
+              if(sfc->source()->center_vertex() ==  v){
+                v_vertex_of_f = true;
+                break;
```

Comment:
This will only break the inner loop. Perhaps if re-written as a function, a `return true` here would be better?

Reply:
I appreciate your careful reading of the code.   I just pushed a fix.